sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/util/ApplyIfChanged.kt

```diff
@@ -0,0 +1,5 @@
+package io.sentry.kotlin.multiplatform.util
+
+internal inline fun <T> applyIfChanged(old: T, new: T, applyNew: (T) -> Unit) {
+    if (old != new) applyNew(new)
```

Comment:
hm, how does this work for non-primitive types? E.g. Breadcrumbs list or User? They have to be `data classes` or implement `equals` right?

Reply:
yes good point, I'll double check which fields are affected by this

Reply:
in any case we need some way to detect whether a value changed or not and only then apply it